### PR TITLE
feat: `cast send --create`

### PR DESCRIPTION
**Pull request by gakonst**
Previously, `cast send` would not allow for an optional `to`, which would preclude people from deploying raw bytecode. This addresses it in the same way that `cast estimate --create` works. Also refactors the `cast send` logic to a separate file.

### CODE REVIEW

**Comment by stong** [comment]
Works like a charm! Thank you @gakonst

File: cli/src/cmd/cast/send.rs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    etherscan_api_key: Option<String>,
+    cast_async: bool,
+    confs: usize,
+    to_json: bool,
+) -> eyre::Result<()>
+where
+    M::Error: 'static,
+{
+    let sig = args.0;
+    let params = args.1;
```
Comment (rkrasiuk) [inline cli/src/cmd/cast/send.rs:235]: nit: 
```suggestion
    let (sig, params) = args;
```

**Review by rkrasiuk — APPROVED** [review]
lgtm

File: cli/src/cmd/cast/send.rs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                        cast_async,
+                        confirmations,
+                        to_json,
+                    )
+                    .await?;
+                }
+            } // Checking if signer isn't the default value
+              // 00a329c0648769A73afAc7F9381E08FB43dBEA72.
+        } else if config.sender !=
+            Address::from_str("00a329c0648769A73afAc7F9381E08FB43dBEA72").unwrap()
```
[the remark this answers is not in the document]
Comment (joshieDo) [reply cli/src/cmd/cast/send.rs:177]: `Config::DEFAULT_SENDER`

[the remark this answers is not in the document]
Comment (mattsse) [reply cli/src/cmd/cast/send.rs:177]: yeah good idea,
can do this in a followup since this is used in a bunch of places

**Review by mattsse — APPROVED** [review]

**Comment by gakonst** [comment]
thank you sirs

**Comment by syspulse** [comment]
Could anyone provide and example how to use it ?
`
cast -V
cast 0.2.0 (d94e3c6 2024-04-01T00:19:33.859963491Z)

cast send --create 0x60806040526004600655348 "constructor()" --private-key $SK
error: unexpected argument '--private-key' found

cast send --create 0x60806040526004600655348 --private-key $SK
error: unexpected argument '--private-key' found
`

**Comment by arr00** [comment]
> Could anyone provide an example how to use it ?
> 
> ```
> cast -V
> cast 0.2.0 (d94e3c6 2024-04-01T00:19:33.859963491Z)
> 
> cast send --create 0x60806040526004600655348 "constructor()" --private-key $SK
> error: unexpected argument '--private-key' found
> 
> cast send --create 0x60806040526004600655348 --private-key $SK
> error: unexpected argument '--private-key' found
> ```

Additional args must come before the bytecode and the bytecode must immediately follow the `--create` flag. The following will work:
```
cast send --private-key $SK --create 0x60806040526004600655348 
```

**Pull request merged**